src/librustdoc/html/static/main.js

```diff
@@ -1664,6 +1664,18 @@ function defocusSearchBar() {
             var ret_in_args = addTab(results.in_args, query, false);
             var ret_returned = addTab(results.returned, query, false);
 
+            if ((currentTab === 0 && ret_others[1] === 0) ||
```

Comment:
Can we add some comments on the flow of user behavior above this?
```suggestion
            // Navigate to the relevant tab if the current tab is empty,
            // like it users search for "-> String". If they had selected
            // another tab previously, they have to click it again.
            if ((currentTab === 0 && ret_others[1] === 0) ||
```

Reply:
I strongly approve, thanks!